packages/webapp/src/components/Form/Unit/index.jsx

```diff
@@ -155,156 +45,61 @@ const Unit = ({
+    onChangeUnit,
     reactSelectWidth,
-  } = useMemo(() => {
-    const databaseUnit = defaultValueUnit ?? unitType.databaseUnit;
-    const options = getOptions(unitType, system);
-    const hookFormValue = hookFormGetValue(name);
-    const value = hookFormValue || (hookFormValue === 0 ? 0 : defaultValue);
-    const isSelectDisabled = options.length <= 1;
-    const measure = convert().describe(databaseUnit)?.measure;
-    const reactSelectWidth = getReactSelectWidth(measure);
-    return to && convert().describe(to)?.system === system
-      ? {
-          displayUnit: to,
-          displayValue: defaultValue && roundToTwoDecimal(convert(value).from(databaseUnit).to(to)),
-          options,
-          databaseUnit,
-          isSelectDisabled,
-          measure,
-          reactSelectWidth,
-        }
-      : {
-          ...getDefaultUnit(unitType, value, system, databaseUnit),
-          options,
-          databaseUnit,
-          isSelectDisabled,
-          measure,
-          reactSelectWidth,
-        };
-  }, []);
-  const reactSelectStyles = useReactSelectStyles(disabled, { reactSelectWidth });
-
-  const hookFormUnitOption = hookFromWatch(displayUnitName);
-  const hookFormUnit = databaseUnit;
-  useEffect(() => {
-    if (typeof hookFormUnitOption === 'string' && getUnitOptionMap()[hookFormUnitOption]) {
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[hookFormUnitOption]);
-    }
-  }, []);
-  useEffect(() => {
-    if (hookFormUnit && convert().describe(hookFormUnit)?.system !== system && measure !== 'time') {
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[displayUnit]);
-    }
-  }, [hookFormUnit]);
-
-  useEffect(() => {
-    !hookFormGetValue(displayUnitName) &&
-      hookFormSetValue(displayUnitName, getUnitOptionMap()[displayUnit]);
-  }, []);
-
-  const [visibleInputValue, setVisibleInputValue] = useState(displayValue);
-  const hookFormValue = hookFromWatch(name, defaultValue);
-
-  useEffect(() => {
-    hookFormSetHiddenValue(hookFormValue, { shouldValidate: true, shouldDirty: false });
-  }, []);
-
-  useEffect(() => {
-    if (hookFormUnit && hookFormValue !== undefined) {
-      setVisibleInputValue(
-        roundToTwoDecimal(convert(hookFormValue).from(databaseUnit).to(hookFormUnit)),
-      );
-      //Trigger validation
-      (hookFormValue === 0 || hookFormValue > 0) && hookFormSetHiddenValue(hookFormValue);
-    }
-  }, [hookFormUnit]);
-
-  const inputOnChange = (e) => {
-    setVisibleInputValue(e.target.value);
-    mode === 'onChange' && inputOnBlur(e);
-  };
+    dividerWidth,
+  } = useUnit({
+    disabled,
+    name,
+    displayUnitName,
+    hookFormSetValue,
+    hookFormGetValue,
+    hookFromWatch,
+    defaultValue,
+    system,
+    control,
+    unitType,
+    from: defaultValueUnit,
+    to,
+    required,
+    optional,
+    mode,
+    max,
+    onBlur,
+    onChangeUnitOption,
+    autoConversion,
+  });
 
-  const hookFormSetHiddenValue = useCallback(
```

Comment:
@SayakaOno sorry I am just tagging you here so I can ask you a question about it beyond this PR